homeassistant/components/enigma2/media_player.py

```diff
@@ -226,38 +165,39 @@ async def async_mute_volume(self, mute: bool) -> None:
         """Mute or unmute."""
         await self._device.toggle_mute()
 
-    @property
-    def source(self):
-        """Return the current input source."""
-        return self._device.status.currservice.station
-
-    @property
-    def source_list(self):
-        """List of available input sources."""
-        return self._device.source_list
-
     async def async_select_source(self, source: str) -> None:
         """Select input source."""
         await self._device.zap(self._device.sources[source])
 
     async def async_update(self) -> None:
         """Update state of the media_player."""
         await self._device.update()
-
-    @property
-    def extra_state_attributes(self):
-        """Return device specific state attributes.
-
-        isRecording:        Is the box currently recording.
-        currservice_fulldescription: Full program description.
-        currservice_begin:  is in the format '21:00'.
-        currservice_end:    is in the format '21:00'.
-        """
-        if self._device.status.in_standby:
-            return {}
-        return {
-            ATTR_MEDIA_CURRENTLY_RECORDING: self._device.status.is_recording,
-            ATTR_MEDIA_DESCRIPTION: self._device.status.currservice.fulldescription,
-            ATTR_MEDIA_START_TIME: self._device.status.currservice.begin,
-            ATTR_MEDIA_END_TIME: self._device.status.currservice.end,
-        }
+        self._attr_available = not self._device.is_offline
+        self._attr_media_title = self._device.status.currservice.station
+        self._attr_media_series_title = self._device.status.currservice.name
+        self._attr_media_channel = self._device.status.currservice.station
+        self._attr_is_volume_muted = self._device.status.muted
+        self._attr_media_content_id = self._device.status.currservice.serviceref
+        self._attr_media_image_url = self._device.picon_url
+        self._attr_source = self._device.status.currservice.station
+        self._attr_source_list = self._device.source_list
+        self._attr_state = (
+            MediaPlayerState.OFF
+            if self._device.status.in_standby
+            else MediaPlayerState.ON
+        )
+        self._attr_volume_level = (
+            self._device.status.volume / 100
+            if self._device.status.volume is not None
+            else None
+        )
+        self._attr_extra_state_attributes = (
+            {
+                ATTR_MEDIA_CURRENTLY_RECORDING: self._device.status.is_recording,
+                ATTR_MEDIA_DESCRIPTION: self._device.status.currservice.fulldescription,
+                ATTR_MEDIA_START_TIME: self._device.status.currservice.begin,
+                ATTR_MEDIA_END_TIME: self._device.status.currservice.end,
+            }
+            if not self._device.status.in_standby
+            else {}
+        )
```

Comment:
Please avoid inline if statements if they take more than one line

Reply:
I have it now as normal if/else statements :slightly_smiling_face: